src/cmap/auth/mongo_credentials.ts

```diff
@@ -132,15 +132,15 @@ export class MongoCredentials {
     ) {
       if (this.source != null && this.source !== '$external') {
         // TODO(NODE-3483): Replace this with a MongoAuthValidationError
-        throw new MongoDriverError(
+        throw new MongoRuntimeError(
           `Invalid source '${this.source}' for mechanism '${this.mechanism}' specified.`
         );
       }
     }
 
     if (this.mechanism === AuthMechanism.MONGODB_PLAIN && this.source == null) {
       // TODO(NODE-3483): Replace this with a MongoAuthValidationError
-      throw new MongoDriverError('PLAIN Authentication Mechanism needs an auth source');
+      throw new MongoRuntimeError('PLAIN Authentication Mechanism needs an auth source');
```

Comment:
API error, someone would need to request PLAIN (LDAP) auth but not specify a db for the authSource